server/lib/wsapi/chat.ts

```diff
@@ -78,22 +80,17 @@ export class ChatApi {
     }),
     'getUser',
     throttleMiddleware(joinThrottle, (data: Map<string, any>) => data.get('user')),
-    'getChannel',
   )
   async join(data: Map<string, any>) {
     const user = data.get('user')
-    const channel = data.get('channel')
-    if (this.state.users.has(user.name) && this.state.users.get(user.name)!.has(channel)) {
-      throw new errors.BadRequest('already in this channel')
-    }
-
-    await addUserToChannel(user.session.userId, channel)
+    const { channel } = data.get('body')
 
-    this.state = this.state
-      .updateIn(['channels', channel], (s = Set()) => s.add(user.name))
-      .updateIn(['users', user.name], (s = Set()) => s.add(channel))
-    this._publishTo(channel, { action: 'join', user: user.name })
-    this._subscribeUserToChannel(user, channel)
+    try {
+      const chatService = container.resolve(ChatService)
```

Comment:
Just a note (you don't need to fix this for this PR because I know this is moving to HTTP after this *anyway*), my advice before was only to not resolve this *at the top level*. It's perfectly fine to inject these into an instance of something and then use it as a field